src/hooks/use-path-validation.ts

```diff
@@ -0,0 +1,77 @@
+import { useEffect, useState } from 'react'
+import { useHistory } from 'react-router'
+import { TCR_EXISTENCE_TEST } from 'utils/graphql'
+import { ApolloClient, InMemoryCache } from '@apollo/client'
+import { HttpLink } from '@apollo/client/link/http'
+
+const usePathValidation = () => {
+  const history = useHistory()
+  const [pathResolved, setPathResolved] = useState<boolean>(false)
+  const [invalidTcrAddr, setInvalidTcrAddr] = useState<boolean>(false)
+
+  useEffect(() => {
+    const checkPathValidation = async () => {
+      const pathname = history.location.pathname
+      const search = history.location.search
+      const isOldPath = /\/tcr\/0x/.test(pathname)
+
+      if (isOldPath) {
+        const searchParams = new URLSearchParams(search)
+        let chainId = null
+        const matches = pathname.match(/tcr\/(0x[0-9a-zA-Z]+)/)
+        const tcrAddress = matches ? matches[1].toLowerCase() : null
+
+        if (searchParams.has('chainId')) chainId = searchParams.get('chainId')
```

Comment:
Don't bother with the `?chainId=...`, you should ignore it now and delete all code that deals with it.